weblate/machinery/google.py

```diff
@@ -99,3 +101,46 @@ def get_error_message(self, exc):
                 pass
 
         return super().get_error_message(exc)
+
+
+class GoogleTranslationAdvanced(MachineTranslation):
+    """Google Translate API v3 machine translation support."""
+
+    name = "Google Translate Advanced"
+    max_score = 90
+
+    def __init__(self):
+        """Check configuration."""
+        super().__init__()
+        if settings.MT_GOOGLE_CREDENTIALS is None:
+            raise MissingConfiguration("Google Translate requires API key")
+        if settings.MT_GOOGLE_PROJECT is None:
+            raise MissingConfiguration(
+                "You have to specify Google Cloud project "
+                "affiliated with provided credentials"
+            )
+        credentials = service_account.Credentials.from_service_account_file(
+            settings.MT_GOOGLE_CREDENTIALS
+        )
+        self.client = TranslationServiceClient(credentials=credentials)
+        self.parent = self.client.location_path(settings.MT_GOOGLE_PROJECT, "global")
+
+    def download_languages(self):
+        """List of supported languages."""
+        return [
+            l.language_code
+            for l in self.client.get_supported_languages(self.parent).languages
+        ]
+
+    def download_translations(self, source, language, text, unit, user):
+        """Download list of possible translations from a service."""
+        trans = self.client.translate_text(
+            [text], language, self.parent, source_language_code=source
+        )
+
+        yield {
+            "text": trans.translations[0].translated_text,
```

Comment:
Are there more translations returned? In that case it might make sense to return them all.

Reply:
Given that the method is fed by text, it returns one translation. Yes, there is possibility to translate multiple strings on one call, yet i do not know where to get all the string to translate. Should i override some machine function than?

Reply:
Okay, in case it matches number of strings passed to it, then it can stay as it is. We currently do not have batch API in machinery (though it will be probably added in future, see https://github.com/WeblateOrg/weblate/issues/3393).